### PR TITLE
Сreating a user access system through a role group

### PR DESCRIPTION
**Pull request by IvanYukish**
Creating an application permissions system. To add permissions to your directory, add PermissionPequiredMixin and select the appropriate permission.

### CODE REVIEW

File: feedbacks/tests.py [hunk 1]
```diff
@@ -42,17 +42,16 @@ def test_call_view_denies_anonymous(self):
     def test_call_view_loads(self):
         self.client.login(username='user', password='test')
         response = self.client.get('/feedbacks/')
-        self.assertEqual(response.status_code, 200)
-        self.assertTemplateUsed(response, 'feedbacks/feedback_list.html')
+        self.assertEqual(response.status_code, 403)
```
Comment (MrKreg) [inline feedbacks/tests.py]: Remove permissions from view instead of change tes

**Review by MrKreg — CHANGES_REQUESTED** [review]

File: quiz/views.py [hunk 2]
```diff
@@ -1,4 +1,5 @@
-from django.contrib.auth.mixins import LoginRequiredMixin
+from django.contrib.auth.mixins import LoginRequiredMixin, \
```
Comment (falled10) [inline quiz/views.py]: You have unresolved import LoginRequiredMixin.

File: users/views.py [hunk 3]
```diff
@@ -1,19 +1,25 @@
...
+    PermissionRequiredMixin, UserPassesTestMixin
 from django.shortcuts import render, redirect
 from django.urls import reverse
-from django.views.generic import CreateView, UpdateView
-
-from .forms import UserRegisterForm, UserUpdateForm
+from django.views.generic import CreateView, UpdateView, ListView, DeleteView
+from django.contrib.auth.models import Group
+from django.urls import reverse_lazy
+from .forms import UserRegisterForm, UserUpdateForm, GroupForm
```
Comment (falled10) [inline users/views.py]: You should have blank line before local imports, see [pep8](https://www.python.org/dev/peps/pep-0008/#imports)

File: feedbacks/tests.py [hunk 4]
```diff
@@ -43,7 +43,7 @@ def test_call_view_loads(self):
         self.client.login(username='user', password='test')
         response = self.client.get('/feedbacks/')
         self.assertEqual(response.status_code, 200)
-        self.assertTemplateUsed(response, 'feedbacks/feedback_list.html')
+        self.assertTemplateUsed(response, "feedbacks/feedback_list.html")
```
Comment (MrKreg) [inline feedbacks/tests.py]: Can you explain why did you change this row?

File: indicators/views.py [hunk 5]
```diff
@@ -1,7 +1,8 @@
 from django.views.generic import CreateView, ListView, DeleteView, UpdateView
 from django.contrib.messages.views import SuccessMessageMixin
 from django.urls import reverse_lazy
-from django.contrib.auth.mixins import LoginRequiredMixin
+from django.contrib.auth.mixins import LoginRequiredMixin, \
```
Comment (MrKreg) [inline indicators/views.py]: Can you follow team agreements about multiple import with scopes?

File: users/views.py [hunk 6]
```diff
@@ -84,3 +90,54 @@ def get_form_kwargs(self):
         kwargs = super(PasswordChangeView, self).get_form_kwargs()
         kwargs['user'] = kwargs.pop('instance')
         return kwargs
+
+
+class ListGroups(LoginRequiredMixin, PermissionRequiredMixin, ListView):
+    model = Group
+    context_object_name = 'group'
+    template_name = 'users/group.html'
+    queryset = Group.objects.all()
```
Comment (falled10) [inline users/views.py]: You use ```model = Group``` and ```queryset = Group.objects.all()``` can you use only one of them?

File: users/views.py [hunk 7]
```diff
@@ -84,3 +90,54 @@ def get_form_kwargs(self):
...
+        return context
+
+
+class DeleteGroups(LoginRequiredMixin, PermissionRequiredMixin, DeleteView):
+    template_name = 'users/delete_Group.html'
+    context_object_name = 'group'
+    model = Group
+    form_class = GroupForm
+    success_url = reverse_lazy('group_perm-list')
+    success_message = 'Group: "%(name)s" was deleted successfully'
```
Comment (falled10) [inline users/views.py:129]: You have success_message here but have not SuccessMessageMixin, is it right?

File: users/views.py [hunk 8]
```diff
@@ -84,3 +90,54 @@ def get_form_kwargs(self):
...
+    permission_required = 'auth.delete_group'
+
+
+class CreateGroup(LoginRequiredMixin, PermissionRequiredMixin, CreateView):
+    model = get_user_model()
+    form_class = GroupForm
+    context_object_name = 'group'
+    template_name = 'users/group_permissions.html'
+    success_url = reverse_lazy('group_perm-list')
+    success_message = 'Country indicator: "%(name)s" was created successfully'
```
Comment (falled10) [inline users/views.py]: You have success_message here but have not SuccessMessageMixin, is it right?

**Review by falled10 — CHANGES_REQUESTED** [review]

**Review by MrKreg — CHANGES_REQUESTED** [review]

**Review by MrKreg — DISMISSED** [review]

**Review by falled10 — APPROVED** [review]

**Review by kolinkovlad — APPROVED** [review]

**Review by rostIvan — APPROVED** [review]

**Pull request merged**